utils.js

```diff
@@ -0,0 +1,124 @@
+const _ = require('lodash')
+const pugLexer = require('pug-lexer')
+const pugParser = require('pug-parser')
+const pugWalk = require('pug-walk')
+const vfile = require('vfile')
+const vfileLocation = require('vfile-location')
+
+// https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+const JS_MINE = [
+  'application/javascript',
+  'application/ecmascript',
+  'text/ecmascript',
+  'text/javascript',
+]
+
+let ctx
```

Comment:
Is there no better way than a global?

Reply:
This global is the same as `var extractions = []` in original `index.js`. I know this is not good enough method. But I have no idea about that.

Reply:
haha I didn't notice that (because of the split files)